src/main/java/Duke.java

```diff
@@ -1,49 +1,169 @@
+import java.io.IOException;
 import java.util.Scanner;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Objects;
 
 public class Duke {
     public static final int DEADLINE_STRING_LENGTH = 9;     // "deadline "
     public static final int BY_SEPARATOR_LENGTH = 4;        // "/by  "
     public static final int EVENT_STRING_LENGTH = 6;        // "event "
     public static final int AT_SEPARATOR_LENGTH = 4;        // "/at  "
+    public static final String DATA_FILE_PATH = "data.txt";
     public static ArrayList<Task> list = new ArrayList<Task>();
 
-    public static void printList(ArrayList<Task> list) {
-            System.out.println("        ____________________________________________");
-            for (int i = 1; i <= list.size(); i += 1) {
-                String taskType = list.get(i-1).getTaskType();
-                String markDoneStatus = list.get(i-1).isDone()?"[X]":"[ ]";
-
-                switch(taskType) {
-                case "ToDo":
-                    System.out.println("        [T]" + markDoneStatus + " " + i + ". " + list.get(i-1).getTaskName());
-                    break;
-
-                case "Deadline":
-                    String deadline = list.get(i-1).getTaskDeadline();
-                    System.out.println("        [D]" + markDoneStatus + " " + i + ". " + list.get(i-1).getTaskName() +
-                            " (by: " + deadline + ")");
-                    break;
-
-                case "Event":
-                    String eventTime = list.get(i-1).getEventTime();
-                    System.out.println("        [E]" + markDoneStatus + " " + i + ". " + list.get(i-1).getTaskName() +
-                            " (at: " + eventTime + ")");
-                    break;
-
-                default:
-                    break;
+    public static void initializeFile() {
+        File f = new File(DATA_FILE_PATH);
+        if (!f.exists()) {
+            try {
+                f.createNewFile();
+            } catch (IOException e) {
+                System.out.println("An error occurred while creating 'data.txt'");
+            }
+        }
+    }
+
+    public static void populateInitialList(String filePath) {
+        File f = new File(DATA_FILE_PATH);
+        Scanner s = null;
+        try {
+            s = new Scanner(f);
+            s.useDelimiter(System.lineSeparator());
+        } catch (FileNotFoundException e) {
+            System.out.println("'data.txt' not found");
+        }
+
+        while (s.hasNext()) {
+            String nextTask = s.next();
+            String[] taskParameters = nextTask.split(" \\| ");
+            Boolean taskStatus = null;
+
+            switch (taskParameters[0]) {
+            case "T":
+                if (Objects.equals(taskParameters[1], "1")) {
+                    taskStatus = true;
+                } else if ((Objects.equals(taskParameters[1], "0"))) {
+                    taskStatus = false;
+                }
```

Comment:
Perhaps this could be more readable?
```suggestion
                taskStatus = Objects.equals(taskParameters[1], "1");
```